scripts/safe-commit-linter.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+set -e
+
+###############################################################################
+#
+#   safe-commit-linter.sh
+#
+#   Executes safe_template_linter.py on the set of files in a particular git
+#   commit.
+#
+###############################################################################
+
+show_help() {
+    echo "Usage: safe-commit-linter.sh [OPTION]"
+    echo "Runs the Safe Template Linter against all files in a git commit."
+    echo ""
+    echo "Mandatory arguments to long options are mandatory for short options too."
+    echo "  -c, --commit=COMMIT         Run against the provided commit. Defaults"
+    echo "                              to the latest commit of the current branch."
+}
+
+for i in "$@"; do
+    case $i in
+        -c=*|--commit=*)
+            COMMIT="${i#*=}"
+            shift # past argument=value
+            ;;
+        -h|--help|*)
+            # help or unknown option
+            show_help
+            exit 0
+            ;;
+    esac
+done
+
+current_branch_hash=`git rev-parse HEAD`
+if [ -z "${COMMIT+x}" ]; then
+    # if commit is not set, get hash of current branch
+    COMMIT="$current_branch_hash"
+else
+    # verify that the current branch has the commit to be tested
+    commit_hash=`git rev-parse --verify "$COMMIT"`
+    merge_base=`git merge-base "$COMMIT" "$current_branch_hash"`
+
+    if [ "$merge_base" != "$commit_hash" ]; then
+        echo "$COMMIT is not an ancestor of the current branch." 1>&2
+        exit 1
+    fi
+fi
+
+for f in `git show --pretty="format:" --name-only "$COMMIT"`; do
```

Comment:
This is going to show the files only from that single commit, right?  So it won't lint all the files in your changes?